packages/mdc-textfield/_mixins.scss

```diff
@@ -444,70 +447,57 @@
 @mixin mdc-text-field-textarea-disabled_ {
-  @include mdc-text-field-textarea-stroke-color_($mdc-textarea-disabled-border-color);
+  @include mdc-text-field-outlined-disabled_;
   @include mdc-text-field-textarea-fill-color_($mdc-textarea-disabled-background);
-
-  border-bottom-width: 1px;
-  border-style: solid;
-
-  .mdc-text-field__input {
-    border: 1px solid transparent;
-  }
-
-  .mdc-floating-label {
-    @include mdc-floating-label-fill-color($mdc-textarea-disabled-background);
-  }
 }
 
 @mixin mdc-text-field-textarea-invalid_ {
-  @include mdc-text-field-textarea-stroke-color($mdc-text-field-error);
+  @include mdc-text-field-outline-color($mdc-text-field-error);
+  @include mdc-text-field-hover-outline-color($mdc-text-field-error);
+  @include mdc-text-field-focused-outline-color($mdc-text-field-error);
 }
 
 @mixin mdc-text-field-textarea_ {
-  @include mdc-text-field-textarea-shape-radius(small);
-  @include mdc-text-field-textarea-stroke-color($mdc-textarea-border);
-
-  // Translate above the top of the input, and compensate for the amount of offset needed
-  // to position the label within the bounds of the inset padding.
-  @include mdc-floating-label-float-position($mdc-text-field-textarea-label-position-y, 0%, $mdc-text-field-textarea-label-scale);
-  @include mdc-floating-label-shake-animation(textarea);
+  // Note: The refactor to add the filled textarea style will result in a lot of the specificity issues that the
+  // textarea has being resolved by using the same --outlined variant class as the text field.
+  @include mdc-text-field-outline-color($mdc-text-field-outlined-idle-border);
+  @include mdc-text-field-hover-outline-color($mdc-text-field-outlined-hover-border);
+  @include mdc-text-field-focused-outline-color(primary);
+  @include mdc-floating-label-shake-animation(text-field-outlined);
+  @include mdc-text-field-outline-shape-radius(small);
   @include mdc-states-base-color(transparent);
   @include mdc-text-field-fill-color(transparent);
+  @include mdc-floating-label-outlined-float-position($mdc-text-field-outlined-label-position-y, 0%);
 
   $padding-inset: 16px;
   $label-offset-y: $padding-inset + 2;
   $label-offset-x: $padding-inset;
 
-  display: flex;
+  display: inline-flex;
   width: auto;
   height: auto;
   transition: none;
-  border: 1px solid;
-  overflow: hidden;
+  overflow: visible;
 
   .mdc-text-field__input {
     align-self: auto;
-    box-sizing: content-box;
+    box-sizing: border-box;
     height: auto;
-    margin: 0;
-    padding: $padding-inset;
-    padding-top: $padding-inset * 2;
-    border: 1px solid transparent;
+    margin: $padding-inset/2 1px 1px 0;
+    padding: 0 $padding-inset $padding-inset;
+    border: none;
   }
 
   .mdc-floating-label {
-    @include mdc-floating-label-fill-color($mdc-textarea-background);
-    @include mdc-rtl-reflexive(left, 1px, right, 0);
-    @include mdc-rtl-reflexive-property(margin, 8px, 0);
-
-    top: $label-offset-y;
+    top: 17px;
```

Comment:
This was formerly using `$label-offset-y` but is now hard-coded, and the `$label-offset-x`/`y` variables are now unused.